compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -31,25 +31,41 @@
 #include "stmt.h"
 #include "passes.h"
 
-/* Print lots of debugging messages when DEBUG is defined */
-#ifdef DEBUG
-  #define DEBUG_PRINTF(...) printf(__VA_ARGS__)
-#else
-  #define DEBUG_PRINTF(...)
-#endif
+// There are two questions:
+// 1- is the primitive/function eligible to run in a fast block?
+//    any blocking or system call disqualifies it since a fast
+//    AM handler can be run in a signal handler
+// 2- is the primitive/function communication free?
+
+// Any function containing comminication can't run in a fast block.
+
+enum {
+  // The primitive is ineligible for a fast (e.g. uses a lock or allocator)
+  // AND it causes communication
+  NOT_FAST_NOT_LOCAL,
+  // Is the primitive ineligible for a fast (e.g. uses a lock or allocator)
+  // but communication free?
+  LOCAL_NOT_FAST,
```

Comment:
Really minor, but it seems like this should be `NOT_FAST_LOCAL` to match the others